src/js/hooks/list-pages/productSupplier/useProductSupplierActions.jsx

```diff
@@ -0,0 +1,90 @@
+import React, { useCallback } from 'react';
+
+import { confirmAlert } from 'react-confirm-alert';
+import { RiDeleteBinLine, RiPencilLine } from 'react-icons/ri';
+import { useSelector } from 'react-redux';
+
+import productSupplierApi from 'api/services/ProductSupplierApi';
+import { PRODUCT_SUPPLIER_URL } from 'consts/applicationUrls';
+import CustomConfirmModal from 'utils/CustomConfirmModal';
+import { hasPermissionsToProductSourceActions } from 'utils/permissionUtils';
+
+import 'react-confirm-alert/src/react-confirm-alert.css';
+
+const useProductSupplierActions = ({ fireFetchData }) => {
+  const { currentUser, isAdmin } = useSelector((state) => ({
+    currentUser: state.session.user,
+    isAdmin: state.session.isUserAdmin,
+  }));
+
+  const deleteProductSupplier = async (onClose, productSupplierId) => {
+    try {
+      await productSupplierApi.deleteProductSupplier(productSupplierId);
+      fireFetchData?.();
+    } finally {
+      onClose?.();
+    }
+  };
+
+  const modalLabels = {
+    title: {
+      label: 'react.productSupplier.deleteConfirmation.title.label',
+      default: 'Are you sure?',
+    },
+    content: {
+      label: 'react.productSupplier.deleteConfirmation.content.label',
+      default: 'Are you sure you want to delete this Product Source?',
+    },
+  };
+
+  const deleteConfirmationModalButtons = (onClose, productSupplierId) => ([
+    {
+      variant: 'transparent',
+      defaultLabel: 'Cancel',
+      label: 'react.productSupplier.deleteConfirmation.cancel.label',
+      onClick: onClose,
+    },
+    {
+      variant: 'danger',
+      defaultLabel: 'Delete',
+      label: 'react.productSupplier.deleteConfirmation.delete.label',
+      onClick: () => deleteProductSupplier(onClose, productSupplierId),
+    },
+  ]);
+
+  const openConfirmationModal = (productSupplierId) => {
+    confirmAlert({
+      customUI: ({ onClose }) => (
+        <CustomConfirmModal
+          labels={modalLabels}
+          onClose={onClose}
+          buttons={deleteConfirmationModalButtons(onClose, productSupplierId)}
+        />
+      ),
+    });
+  };
```

Comment:
We can go one step further and implement a promise based confirm dialog which returns boolean values based on clicked button, I implemented something similar in this PR https://github.com/openboxes/openboxes/pull/4454#issue-2081795239

Let me explain...
There is a javascript confirmation modal build in the browser, to call it we do
```js
const isConfirmed = window.confirm("Are you sure?)
``` 
upon calling it stalls the application until the user responds by confirming or canceling the confirmation dialog.
We can implement something similar with Promises.
The code would look something like

```js

const confirm = ({ modalLabels }) => new Promise((resolve) => confirmAlert({
 willUnmount: () => resolve(false),
  customUI: ({ onClose }) => (
   <CustomConfirmModal
      labels={modalLabels}
      onClose={ () => resolve(false)}
      buttons={{
        variant: 'transparent',
        defaultLabel: 'Cancel',
        label: 'react.productSupplier.deleteConfirmation.cancel.label',
        onClick:  () => resolve(false),
      },
      {
        variant: 'danger',
        defaultLabel: 'Delete',
        label: 'react.productSupplier.deleteConfirmation.delete.label',
        onClick: () => resolve(true),
      }}
    />
    ),
}));
```
and then we use it the following way
```jsx
const isConfirmed = await confirm({ modalLabels })

```


You might say that this restricts us to having the confirmationAlert have only two options since it will return either `true` or `false`. That is true and I think that's the point, and looking through our project we have been using confirmationAlert only with two buttons.

Additionally I think labels for Yes and Cancel could be default labels and have the ability to change them like
```jsx
const isConfirmed = await confirm({ modalLabels, yesLabel: "new yes label" , cancelLabel: "new cancel label" })

```

This is an idea and not a must, but since we are at the point of creating a new confirmation alert component wanted to use this as an opportunity to implement it the bets way possible.
What do you think guys @awalkowiak @alannadolny @kchelstowski ?

Reply:
Regarding the promised-based, I see the idea, but I do not see a case at the moment, where it could be useful. 
Personally I like passing proper `onClick` actions to the `confirmAlert`, it's pretty easy to read.
Additionally, I'd say that the current version also "holds" the application until you click a button, because you can't click anything else while having the modal opened. 

As for the `yesLabel`, `cancelLabel`, we need to be aware that it doesn't always have to be `Yes`/`Cancel`. Even now, we have Cancel/Delete, not Cancel/Yes.

Reply:
it could be _confirm_ and _cancel_, I just noticed in our project 90% of the time we used Yes or No labels.
also in this case Cancel/Delete is basically cancel/Yes logically speaking. The question states _Are you sure you want to delete this Product Source?_ where delete button corresponds to confirming this actions.
Maybe a better an for it would be `confirmLabel` instead of `yesLabel`

Reply:
one of the usecases for readability for using the promised-based confirmationAlert would be code repetition.
An example is addItems page where we make sure the user wants to continue if the form is invalid

with callbacks
```jsx
if (invalid) {
  confirmationAler({
   buttons: {
     label: "Yes",
    onClick: () => callMethodA();
    },
   buttons: {
     label: "No",
    }
  })
} else {
callMethodA()
}
```

with promises
```jsx
if(invalid) {
  const shoudlContinue = await confirmAler();
  if (!shoudlContinue) {
    return;
  }
}
 callMethodA()
```

but this is just my preference, I like reading synchounouse code you probably like callbacks :smile: 
As I stated, this was just a suggestion and a idea submission not a change request for Alan

Reply:
It actually looks interesting, but I would still need to see it "on a real example", e.g. here, but the question is if we still want to "torture" Alan in this PR. 
If everyone gives his opinion, we can make a decision. I'm neither against nor pro implementing this now (in this PR).

Reply:
Yeah, we were using only two buttons, but I implemented the custom modal (passed as a customUi prop) to have the ability to have more than two buttons (and each of them is customizable), so I am not pro this second change.